airflow/plugins_manager.py

```diff
@@ -16,17 +15,18 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-
-import imp
+"""Manages all plugins."""
+# noinspection PyDeprecation
+import imp  # pylint: disable=deprecated-module
 import inspect
 import os
 import re
-from typing import Any, List
+import sys
+from typing import Any, Callable, List, Optional
 
 import pkg_resources
 
 from airflow import settings
-from airflow.models.baseoperator import BaseOperatorLink
```

Comment:
We're removing this one and the tighter type integartion because otherwise this pulls in all of airflow.models which we want to avoid?

Is this worth using an `if TYPING:` for?

Reply:
I am strongly agains using those kind of hacks - they are usually a symptom of an entangled architecture and not best choice of packages/files to split the code. It's like a "broken window" policy - once we let them in, then it's easier to add another exception. So I would like to avoid it at all cost.

However in this case I have an idea that maybe we will be able to implement in the better version of plugins_manager unentangling. I will try it shortly.